legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java

```diff
@@ -16,42 +16,39 @@
 
 package org.finos.legend.engine.testData.generation.service;
 
-import org.eclipse.collections.api.RichIterable;
 import org.finos.legend.engine.language.pure.compiler.toPureGraph.HelperValueSpecificationBuilder;
 import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
 import org.finos.legend.engine.protocol.pure.v1.model.data.EmbeddedData;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.data.RelationalCSVData;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.data.RelationalCSVTable;
 import org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.raw.Lambda;
-import org.finos.legend.pure.generated.Root_meta_pure_runtime_Runtime;
+import org.finos.legend.pure.generated.Root_meta_relational_metamodel_data_RelationalCSVData;
 import org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping;
 import org.finos.legend.pure.generated.core_relational_relational_testDataGeneration_testDataGeneration;
 import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.LambdaFunction;
-import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.Column;
-import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.relation.Table;
 
 import java.util.Collections;
 import java.util.List;
 
 public class TestDataGenerationService
 {
-    public static List<EmbeddedData> generateEmbeddedData(Lambda query, Root_meta_pure_runtime_Runtime runtime, Mapping mapping, PureModel pureModel)
+    public static List<EmbeddedData> generateEmbeddedData(Lambda query, Mapping mapping, PureModel pureModel)
     {
-        RichIterable<? extends Table> tables = core_relational_relational_testDataGeneration_testDataGeneration.Root_meta_relational_testDataGeneration_getTableFromQuery_FunctionDefinition_1__Mapping_1__Runtime_1__Table_MANY_(
-                buildPureLambda(query, pureModel), mapping, runtime, pureModel.getExecutionSupport());
-        if (tables.isEmpty())
+        Root_meta_relational_metamodel_data_RelationalCSVData relationalCSVData = core_relational_relational_testDataGeneration_testDataGeneration.Root_meta_relational_testDataGeneration_getRelationalCSVDataFromQuery_FunctionDefinition_1__Mapping_1__RelationalCSVData_1_(
+                buildPureLambda(query, pureModel), mapping, pureModel.getExecutionSupport());
+        if (relationalCSVData == null)
         {
             return null;
         }
-        List<RelationalCSVTable> relationalCSVTables = tables.collect(table ->
+        RelationalCSVData data = new RelationalCSVData();
```

Comment:
shouldn't we be calling transformEmbeddedData to transform the results to the Pure protocol and then sending that back?

Reply:
May I know where `transformEmbeddedData` is defined? I haven't found any transform functions in the protocol yet.

Reply:
meta::protocols::pure::vX_X_X::transformation::fromPureGraph::runtime::transformEmbeddedData

Reply:
Thanks. but..This function is not open sourced yet.